Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberViewModel.swift

```diff
@@ -32,13 +33,20 @@ extension CardNumberViewModel: CardNumberViewModelProtocol {
   func validate(cardNumber rawText: String) -> Constants.Bundle.SchemeIcon? {
     let cardNumber = cardUtils.removeNonDigits(from: rawText)
 
-    switch cardValidator.validate(cardNumber: cardNumber) {
-    case .failure, .success(.unknown):
-      delegate?.update(cardNumber: cardNumber, scheme: .unknown)
-      return nil
-    case .success(let scheme):
-      delegate?.update(cardNumber: cardNumber, scheme: scheme)
-      return Constants.Bundle.SchemeIcon(scheme: scheme)
+    switch cardValidator.validateCompleteness(cardNumber: cardNumber) {
+    case .failure:
+        return nil
+    // An unknown scheme can also be generated when an eager validation was matched
+    //    but the number is not complete.
+    // In this case we only update delegate if we have a final result to avoiding
+    //    overriding the eager validation with a less informative update
+    case .success((let isComplete, let scheme)):
+      if isComplete {
```

Comment:
Well spotted move from if to guard, totally agree with you 👍 

 

Not sure I necessarily share the feeling that one line guard exits are good for readability.

In my opinion the separate line for the return early actually 
<img width="460" alt="Screenshot 2022-07-29 at 18 06 45" src="https://user-images.githubusercontent.com/102028170/181809535-f766f46b-78c4-4b26-a2c7-26105f4848a2.png">

<img width="474" alt="Screenshot 2022-07-29 at 18 06 34" src="https://user-images.githubusercontent.com/102028170/181809537-12f63ab1-3cbb-43cd-bc90-c58cdb3c8be0.png">
 
To me, the first image makes it clearly where the early exit is as I navigate the page.
In the second image the code is too compact and its not as quickly clear that it might not all run linearly.

If you feel strongly for the one line, I have no problem to use it though.